pkg/controllers/dnshealthcheckprobe/dnshealthcheckprobe_controller.go

```diff
@@ -87,6 +87,7 @@ func (r *DNSHealthCheckProbeReconciler) Reconcile(ctx context.Context, req ctrl.
 			p.Protocol = protocol
 			p.AdditionalHeaders = previous.Spec.AdditionalHeaders
 			p.ExpectedReponses = previous.Spec.ExpectedReponses
+			p.AllowInsecureCertificate = previous.Spec.AllowInsecureCertificate
```

Comment:
This is setting the flag only when an existing probe is updated, but not when it's newly created. I'd add this also to the `else` block below, line 103

Reply:
nice, thanks, added now.